run_tests/chrome/index.ts

```diff
@@ -0,0 +1,135 @@
+import puppeteer, { Browser, BrowserContext, Page } from "puppeteer";
+import { BrowserNames, puppeteerInteractionTimeout } from "../const";
+import { BrowserDriver } from "../driver";
+import assert from "assert";
+
+let browser: Browser | null;
+let browserContext: BrowserContext | null;
+let mainPage: Page | null;
+
+const assertNotNull = <T>(subject: T | null): T => {
+  assert(subject !== null, `${typeof subject} was null`);
+  return subject;
+};
+
+const uploadDeviceCode = async (
+  deviceNameMatcher: (deviceName: string) => boolean,
+) => {
+  // open espruino tab
+  await assertNotNull(mainPage).locator("#btn_load_code").click();
+  const target = await assertNotNull(browserContext).waitForTarget(
+    (t) => t.url().match(/espruino\.com/) !== null,
+  );
+  const espruinoPage = await target?.page();
+  if (!espruinoPage) {
+    throw "couldn't find loaded espruino page";
+  }
+  espruinoPage.setDefaultTimeout(puppeteerInteractionTimeout);
+  await espruinoPage.bringToFront();
+  // dismiss "welcome" modal
+  await espruinoPage.locator("#guiders_overlay").click();
+  // have to wait for the overlay to transition out 😞
+  espruinoPage.waitForSelector("#guiders_overlay", { hidden: true });
+  // wait a moment for the pages feature detection logic to run
+  await new Promise((res) => setTimeout(res, 500));
```

Comment:
This didn't work for me. I dug into the feature detection logic and the problem is that it uses an async method navigator.bluetooth.getAvailability that isn't guaranteed to return before we click the connect icon.

If we await the same method, the page's feature detection should complete before we click connect:

```
  // wait a moment for the pages feature detection logic to run
  await espruinoPage.waitForFunction('navigator.bluetooth.getAvailability()');
```

Reply:
thanks for digging in here! i also missed an `await` on waiting for the overlay to fade out. fixed both in 1b55678bc772dc36f702bc9f93ae32ec2feb7fdf